lnpbp-0031.md

```diff
@@ -40,8 +42,140 @@ License: CC0-1.0
+data Bool             :: false:0 | true:1
+data Dec              :: zero:48 | one:49 | two:50 | three:51 | four:52 | five:53 | six:54 | seven:55 | eight:56 | nine:57
+data HexDecCaps       :: zero:48 | one:49 | two:50 | three:51 | four:52 | five:53 | six:54 | seven:55 | eight:56 | nine:57 | ten:65 | eleven:66 | twelve:67 | thirteen:68 | fourteen:69 | fifteen:70
+data HexDecSmall      :: zero:48 | one:49 | two:50 | three:51 | four:52 | five:53 | six:54 | seven:55 | eight:56 | nine:57 | ten:97 | eleven:98 | twelve:99 | thirteen:100 | fourteen:101 | fifteen:102
+data U4               :: v0:0 | v1:1 | v2:2 | v3:3 | v4:4 | v5:5 | v6:6 | v7:7 | v8:8 | v9:9 | v10:10 | v11:11 | v12:12 | v13:13 | v14:14 | v15:15
+```
+
+`Bitcoin.sty`, semantic lib id `panel_chamber_ohio_GWzzfwBqzA5BUVER6hqq3rBdDs6UUJ15w1T8ys6WrNr5`:
+```haskell
+typelib Bitcoin -- version v0.10.1-2023.04.20.B
+
+data LockTime         :: U32
+data Outpoint         :: txid Txid, vout Vout
+data Sats             :: U64
+data ScriptBytes      :: [U8 ^ ..0xffffffffffffffff]
+data ScriptPubkey     :: ScriptBytes
+data SeqNo            :: U32
+data SigScript        :: ScriptBytes
+data Tx               :: version TxVer
+                       , inputs [TxIn ^ ..0xffffffffffffffff]
+                       , outputs [TxOut ^ ..0xffffffffffffffff]
+                       , lockTime LockTime
+data TxIn             :: prevOutput Outpoint
+                       , sigScript SigScript
+                       , sequence SeqNo
+                       , witness Witness
+data TxOut            :: value Sats, scriptPubkey ScriptPubkey
+data TxVer            :: U32
+data Txid             :: [U8 ^ 32]
+data Vout             :: U32
+data Witness          :: [[U8 ^ ..0xffffffffffffffff] ^ ..0xffffffffffffffff]
+```
+
+`CommitVerify.sty`, semantic lib id `texas_year_ethnic_CPr8tcdPqWZ3KP8dXNPYavTEkbn8PG7CoJHtfwDFKRHJ`
+```haskell
+typelib CommitVerify -- version v0.10.0-2023.03.05.A
+
+data Message          :: [U8 ^ 32]
+data ProtocolId       :: [U8 ^ 32]
+data MerkleNode       :: [U8 ^ 32]
+
+data TreeNode         :: concealedNode (depth StdLib.U4, hash MerkleNode)
+                       | commitmentLeaf (protocolId ProtocolId, message Message)
+
+data MerkleTree       :: depth StdLib.U4
+                       , entropy U64
+                       , messages {[Byte ^ 32] -> Message}
+                       , map {U16 -> ProtocolId, Message}
+
+data MerkleBlock      :: depth StdLib.U4
+                       , crossSection [TreeNode]
+                       , entropy U64?
```

Comment:
Why `MerkleBlock` entropy is optional while `MerkleTree` one is not?

Reply:
Explained during the call